### DISCUSSION TITLE
Publishing to TestFlight fails due to xcarchive not found error while doing the same from XCode works

### DISCUSSION THREAD

**Discussion by snakebyte3d** [codemagic-ci-cd/codemagic-docs#2224] [Q&A]
### If you are looking for help, please confirm the following...

- [x] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1. Have App Store certificates and profiles configured for the project
2. Configure publishing to App Store internal testing
3. Build Flutter project in --release mode with --flavor production from Codemagic UI with latest XCode, Flutter and Cocoapod versions
4. Error message appears (see below)

### Expected results

1. The archive is published to App Store internal testing without errors
2. While doing the above steps, results should not differ between those obtained in the local environment. We can successfully archive and publish builds from local XCode.

### Actual results

Error message:

"Archive succeeded but the expected xcarchive at /Users/builder/clone/build/ios/archive/MyApp.xcarchive not found
Built /Users/builder/clone/build/ios/archive/MyApp.xcarchive.
Invalid iOS archive. Does not contain Info.plist.

To update the settings, please refer to https://docs.flutter.dev/deployment/ios

Building App Store IPA...                                          402ms
Encountered error while creating the IPA:
error: archive not found at path '/Users/builder/clone/build/ios/archive/MyApp.xcarchive'

Try distributing the app in Xcode: "open /Users/builder/clone/build/ios/archive/MyApp.xcarchive"

### Build id (optional)

6565ba45b1d1a67bba005fca

**Comment by himesh-cm** [comment]
Hi @snakebyte3d, The error message says 
`Invalid iOS archive. Does not contain Info.plist.`
Can you please confirm if your project has the plist file

**Reply by snakebyte3d** [reply]
Yes, I can create the archive and upload it correctly to TestFlight using XCode. No issues. Issues appear only on Codemagic. I am not using the yaml config and I would like to avoid it if possible and use only the UI config.

**Reply by himesh-cm** [reply]
Please create a copy of the workflow, enable iOS for the new workflow, re-run the build, and share the build with me.

**Reply by snakebyte3d** [reply]
Hi Himesh, Not sure how to share the build, but here it is the link with both the 
default workflow: https://codemagic.io/app/63de6709bf7b3bea0d5ff680/build/656c68a466acaa1309b685ce
and the workflow copy: https://codemagic.io/app/63de6709bf7b3bea0d5ff680/build/656c667da0e80d315e7632fa

Now the error does not appear on neither, however even with the success message, the build is not published to TestFlight. I have not changed anything in the settings since the build with the error message:

Running pod install...                                             89.8s
Running Xcode build...                                          
Xcode build done.                                           169.7s
Built /Users/builder/clone/build/ios/iphoneos/MyApp.app.

What I suspect is that there are 2 issues:
1. The build using the UI configuration does not seem to take into consideration the ArchiveAction/customArchiveName from [flavor].xcscheme file as Codemagic was looking for a different archive name than it was specified there
2. Another issue may be related to the fact that MyApp is in fact separated by "." (dot) like "My.App" and Codemagic was looking in fact for "My.xcarchive" instead of "My.App.xcarchive". I have set up the customArchiveName to "My" in Xcode but even this way it was not found by Codemagic.

Obviously "My", "My.App" and "MyApp" are needed to be considered as placeholders so I don't make it public here.

**Reply by himesh-cm** [reply]
The discussion is being continued on the intercom, and developers within the team are looking into it. I will keep the thread posted with updates.

**Comment by icarusdust** [comment]
As the issue is related to Flutter, and can be followed [here](https://github.com/flutter/flutter/issues/140212), I will go ahead and close the thread.
As a workaround, you can try executing `flutter build ios` instead by setting up code signing on Xcode project (`xcode-project use-profiles`) and then run `xcode-project build-ipa`

**Comment by himesh-cm** [comment]
We filed a detailed issue, which you can follow [here](https://github.com/flutter/flutter/issues/140212), and the triage team has already acknowledged it.
​
Meanwhile, with the yaml configuration, You can still build the iOS using the regular process on Codemagic.
​
- setup code-signing as shown [here](https://github.com/codemagic-ci-cd/codemagic-sample-projects/blob/main/ios/ios-native-quick-start/codemagic.yaml#L9)
- flutter build ios
- xcode-project use-profiles
- xcode-project build-ipa
​
You can find the sample yaml [here](https://github.com/codemagic-ci-cd/codemagic-sample-projects/blob/main/ios/ios-native-quick-start/codemagic.yaml#L19) for reference purposes.
